subprojects/scala/src/integTest/groovy/org/gradle/scala/compile/ScalaCompileWithJavaLibraryIntegrationTest.groovy

```diff
@@ -38,4 +38,36 @@ class ScalaCompileWithJavaLibraryIntegrationTest extends AbstractIntegrationSpec
         notExecuted(":lib:processResources", ":lib:jar")
     }
 
+    def "scala and java source directory compilation order can be reversed (task configuration #configurationStyle)"() {
```

Comment:
are there any tests that show you can do this for test source sets or any custom source set?

Reply:
I thought it's not necessary as the code path is the same. But since there is some special handling of the main set in some places, it's probably good to have. I added coverage for a custom source set.